tests/console/validate_first_disk_selection.pm

```diff
@@ -0,0 +1,42 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved. This file is offered as-is,
+# without any warranty.
+
+# Summary: Validate that first disk was selected for installation
+#
+# Maintainer: QA SLE YaST team <qa-sle-yast@suse.de>
+
+use base 'consoletest';
+
+use strict;
+use warnings;
+use testapi;
+
+sub run {
+    select_console 'root-console';
+
+    my @errors;
+    # list info about block devices
+    my @lsblk_output = split(/\n/, script_run('lsblk -n'));
+    # get list of disk names
+    my @disks = map { (split ' ', $_)[0] } grep { /disk/ } @lsblk_output;
+    # get first disk names
+    my $first_disk = shift @disks;
+    # check existing partitioning in first disk
+    push @errors, "First disk $first_disk was not used for partitioning."
+      unless (grep { /$first_disk\d/ } @lsblk_output) > 0;
+    # Check for non-existing partitioning in the rest of the disks
+    for my $disk (@disks) {
+        push @errors, "Disk $disk was wrongly used for partitioning."
```

Comment:
As we might still have some traces from the previous runs in openQA, I would at least add some hints to state that this has to be checked. Otherwise, I would propose to check if any other disks have mount points or are used for swap, as this will mean that they were misused. WDYT?

Reply:
I thought the same, but was not sure if there was too much, in case of those partitionless disks I think you meant, so in those cases disk could also contains something, so now I added this extra check.